sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py

```diff
@@ -426,31 +509,57 @@ def start_recording(
         :rtype: ~azure.communication.callautomation.RecordingProperties
         :raises ~azure.core.exceptions.HttpResponseError:
         """
-        channel_affinity_internal = []
+        ...
 
-        if channel_affinity:
-            for channel in channel_affinity:
-                channel_affinity_internal.append(channel._to_generated(# pylint:disable=protected-access
-                    ))
+    @distributed_trace
+    def start_recording(
+        self,
+        *args: Union['ServerCallLocator', 'GroupCallLocator'],
+        **kwargs
+    ) -> RecordingProperties:
+        # pylint:disable=protected-access
+        channel_affinity: List[ChannelAffinity] = kwargs.pop("channel_affinity", None) or []
+        channel_affinity_internal = [c._to_generated() for c in channel_affinity]
+        call_locator: Optional[CallLocator] = None
+        if args:
+            call_locator = args[0]._to_generated()
+        else:
+            if "call_locator" in kwargs:
+                call_locator = kwargs["call_locator"]._to_generated()
```

Comment:
Oops - I forgot to pop these from kwargs... will update. And might refactor this into a method to reuse between sync and async.